datagen/ridership/process.py

```diff
@@ -15,62 +23,140 @@
+    "Franklin/Foxboro": "CR-Franklin",
+    "Middleborough/Lakeville": "CR-Middleborough",
+    "Lowell": "CR-Lowell",
+    "Haverhill": "CR-Haverhill",
+    "Kingston": "CR-Kingston",
+}
 
-def format_subway_data(path_to_csv_file: str):
+
+def format_ridership_csv(
+    path_to_csv_file: str,
+    date_key: str,
+    route_key: str,
+    count_key: str,
+    route_ids_map: Union[None, Dict[str, str]] = None,
+):
     # read data, convert to datetime
     df = pd.read_csv(path_to_csv_file)
-    df["servicedate"] = pd.to_datetime(df["servicedate"])
+    df[date_key] = pd.to_datetime(df[date_key])
 
     # add holidays
     cal = USFederalHolidayCalendar()
-    holidays = cal.holidays(start=df["servicedate"].min(), end=df["servicedate"].max())
+    holidays = cal.holidays(start=df[date_key].min(), end=df[date_key].max())
 
     # mark as holiday and weekday
-    df["holiday"] = df["servicedate"].dt.date.astype("datetime64").isin(holidays.date)
-    df["weekday"] = df["servicedate"].dt.dayofweek
+    df["holiday"] = df[date_key].dt.date.astype("datetime64").isin(holidays.date)
+    df["weekday"] = df[date_key].dt.dayofweek
 
     # define peak, mark weekdays, convert service date back
     conditions = [(df["holiday"] == False) & (df["weekday"] < 5)]
     choices = ["peak"]
     df["peak"] = np.select(conditions, choices, default="offpeak")
-    df["week"] = df["servicedate"].dt.isocalendar().week
-    df["year"] = df["servicedate"].dt.isocalendar().year
-    df["servicedate"] = df["servicedate"].dt.date.astype(str)
+    df["week"] = df[date_key].dt.isocalendar().week
+    df["year"] = df[date_key].dt.isocalendar().year
+    df[date_key] = df[date_key].dt.date.astype(str)
 
     # select date of the week
     dates = df[df["weekday"] == 0]
-    dates = dates[["servicedate", "week", "year"]].drop_duplicates()
+    dates = dates[[date_key, "week", "year"]].drop_duplicates()
 
     # limit data to just peak, merge back dates
     final = df[df["peak"] == "peak"]
+
     final = (
-        final.groupby(["year", "week", "route_or_line"])["validations"]
+        final.groupby(["year", "week", route_key])[count_key]
         .mean()
         .round()
         .reset_index()
     )
 
     final = final.merge(dates, on=["week", "year"], how="left")
 
-    # get list of bus routes
-    routelist = list(set(final["route_or_line"].tolist()))
+    # get list of routes
+    routelist = list(set(final[route_key].tolist()))
 
     # create dict
     output = {}
 
     # write out each set of routes to dict
     for route in routelist:
-        dftemp = final[final["route_or_line"] == route]
-        dictdata = (
-            dftemp[["servicedate", "validations"]]
-            .rename(columns={"servicedate": "date", "validations": "riders"})
-            .to_dict(orient="records")
-        )
-        output[route] = dictdata
-
+        for_route = final[final[route_key] == route]
+        only_date_and_count = for_route[[date_key, count_key]].dropna()
```

Comment:
We now `.dropna()` to remove `nan` values. We need to do this because there are gaps of >1 week in the CR ridership data that produces `nan` dates after `final.merge(dates)` above.